### PR TITLE
docs: improve self-hosted documentation

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes:

- Remove some TODO's that were accidentally merged
- Replace long sentence with a list

## Context:

Fix up some bad stuff from PR #9303.

Marking this as a draft until we're sure that this is right. 😄 😉 

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/self-hosted-configuration.md [hunk 1]
```diff
@@ -199,7 +199,7 @@ You probably have no need for this option - it is an experimental setting for th
 You can customize the Git author that's used whenever Renovate creates a commit.
 The `gitAuthor` option accepts a RFC5322-compliant string.
 
-If you are migrating from a old Git author to a new Git author, put the old `gitAuthor` into `RENOVATE_LEGACY_GIT_AUTHOR_EMAIL` in environment (TODO: Explain where/what is this "environment"???).
+If you are migrating from a old Git author to a new Git author, put the old `gitAuthor` into `RENOVATE_LEGACY_GIT_AUTHOR_EMAIL` in environment.
```
Comment (HonkingGoose) [inline docs/usage/self-hosted-configuration.md:202]: To me it's not clear what this "environment" is referring to? I guess "environment variable"?

File: docs/usage/self-hosted-configuration.md [hunk 2]
```diff
@@ -242,16 +242,12 @@ Warning: Configuring `logLevel` config option or `--log-level` cli option is dep
 
 ## onboarding
 
-Set this to `false` if (a) you configure Renovate entirely on the bot side (i.e. empty `renovate.json` in repositories) and (b) you wish to run Renovate on every repository the bot has access to, and (c) you wish to skip the onboarding PRs.
```
Comment (HonkingGoose) [inline docs/usage/self-hosted-configuration.md:245]: This sentence is long and hard to understand, so I'd like to replace it with a list.
Please check if the list provided is correct.

File: docs/usage/self-hosted-configuration.md [hunk 3]
```diff
@@ -333,8 +329,6 @@ JSON files will be stored inside the `cacheDir` beside the existing file-based p
 
 Warning: this is an experimental feature and may be modified or removed in a future non-major release.
 
-TODO: Check if this feature is still present in Renovate code.
```
Comment (HonkingGoose) [inline docs/usage/self-hosted-configuration.md:336]: Is `repositoryCache` still present in the Renovate code?

We warn that: "this is an experimental feature and may be modified or removed in a future non-major release."

**Review by HonkingGoose — COMMENTED** [review]
Some things to check before merging. 😉

File: docs/usage/self-hosted-configuration.md (hunk 1 above)
Comment (rarkins) [reply docs/usage/self-hosted-configuration.md:202]: Actually this should be in the new https://docs.renovatebot.com/configuration-options/#gitignoredauthors

File: docs/usage/self-hosted-configuration.md (hunk 2 above)
Comment (rarkins) [reply docs/usage/self-hosted-configuration.md:245]: I can't make suggestions on deleted lines so perhaps you can change the current one into a list and then we edit

File: docs/usage/self-hosted-configuration.md (hunk 3 above)
Comment (rarkins) [reply docs/usage/self-hosted-configuration.md:336]: Yes, it's still present

**Review by rarkins — CHANGES_REQUESTED** [review]

File: docs/usage/self-hosted-configuration.md (hunk 3 above)
Comment (HonkingGoose) [reply docs/usage/self-hosted-configuration.md:336]: Okay, we can keep the `Warning: ....` text then. 😄

File: docs/usage/self-hosted-configuration.md (hunk 1 above)
Comment (HonkingGoose) [reply docs/usage/self-hosted-configuration.md:202]: Will move this. 😄

File: docs/usage/self-hosted-configuration.md [hunk 4]
```diff
@@ -242,16 +239,12 @@ Warning: Configuring `logLevel` config option or `--log-level` cli option is dep
 
 ## onboarding
 
-Set this to `false` if (a) you configure Renovate entirely on the bot side (i.e. empty `renovate.json` in repositories) and (b) you wish to run Renovate on every repository the bot has access to, and (c) you wish to skip the onboarding PRs.
-
 Set this to `false` only if all three statements are true:
 
 - You've configured Renovate entirely on the bot side (e.g. empty `renovate.json` in repositories)
 - You want to run Renovate on every repository the bot has access to
 - You want to skip all onboarding PRs
```
Comment (HonkingGoose) [inline docs/usage/self-hosted-configuration.md:247]: @rarkins I made a list already, and we already merged it via the previous PR... 🙈 😄 

Can you check if this list is correct?

File: docs/usage/configuration-options.md [hunk 5]
```diff
@@ -651,6 +651,9 @@ By default, Renovate will treat any PR as modified if another git author has add
 When a PR is considered modified, Renovate won't perform any further commits such as if it's conflicted or needs a version update.
 If you have other bots which commit on top of Renovate PRs, and don't want Renovate to treat these PRs as modified, then add the other git author(s) to `gitIgnoredAuthors`.
 
+If you are migrating from a old Git author to a new Git author, put the old `gitAuthor` into `RENOVATE_LEGACY_GIT_AUTHOR_EMAIL` in environment.
+Renovate will then check for the old and current Git author before it decides if a branch has been modified.
+
```
Comment (rarkins) [inline docs/usage/configuration-options.md:656]:
```suggestion
```

This is now not needed - use `gitIgnoredAuthors` instead.

File: docs/usage/self-hosted-configuration.md (hunk 4 above)
Comment (rarkins) [reply docs/usage/self-hosted-configuration.md:247]: it's correct

**Review by rarkins — CHANGES_REQUESTED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**